src/arrayMethodJoin.js

```diff
@@ -5,7 +5,32 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let newSeparator = separator;
+
+    if (newSeparator === undefined) {
+      newSeparator = ',';
+    }
+
+    if (newSeparator === null) {
+      newSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let current = this[i];
+
+      if (current === null || current === undefined) {
+        current = '';
+      }
```

Comment:
Instead of checking for a value that is `null` or `undefined`, better check the opposite - not equal to these values. Then you can do `result += current;` directly here.